src/mlpack/methods/kmeans/naive_kmeans_impl.hpp

```diff
@@ -39,47 +39,44 @@ double NaiveKMeans<DistanceType, MatType>::Iterate(const arma::mat& centroids,
   counts.zeros(centroids.n_cols);
 
   // Find the closest centroid to each point and update the new centroids.
-  // Computed in parallel over the complete dataset
   #pragma omp parallel
   {
-    // The current state of the K-means is private for each thread
-    arma::mat localCentroids(centroids.n_rows, centroids.n_cols,
-        arma::fill::zeros);
-    arma::Col<size_t> localCounts(centroids.n_cols, arma::fill::zeros);
+    // Thread-local storage for partial sums
+    arma::mat threadCentroids(centroids.n_rows, centroids.n_cols, arma::fill::zeros);
+    arma::Col<size_t> threadCounts(centroids.n_cols, arma::fill::zeros);
 
-    #pragma omp for
-    for (size_t i = 0; i < (size_t) dataset.n_cols; ++i)
+    #pragma omp for schedule(static) nowait
+    for (size_t i = 0; i < dataset.n_cols; ++i)
     {
       // Find the closest centroid to this point.
-      double minDistance = std::numeric_limits<double>::infinity();
+      double minDistance = std::numeric_limits<double>::max();
```

Comment:
Why make this change? `infinity()` was fine. I mean `max()` works too, but again, better to avoid making changes when possible.